app/services/filter_builder.rb

```diff
@@ -0,0 +1,31 @@
+class FilterBuilder
+  attr_accessor :klass, :params, :results
+
+  def initialize(object_type, params = nil)
+    @klass = object_type[:klass]
+    @params = params
+    @results = nil
+  end
+
+  def perform
+    if @klass == Spell
```

Comment:
This seems like all the functionality of this class relies on the class being passed in `Spell`. We might as well move this to a SpellFilterBuilder class and if we want to use it for the other classes as well we can abstract out what they have in common

Reply:
Makes sense. I created SpellFilterBuilder which just imports from FilterBuilder. So the controller is now written as `SpellFilterBuilder.new(blah ... )`

I don't know how this would work, but does it make sense for the syntax to be FilterBuilder::SpellFilter.new(object, params)? I have seen that before but I do not know how that would be built out.